src/Service/NixUtils.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Nixify\Service;
+
+use function ord;
+
+use const STR_PAD_LEFT;
+
+final class NixUtils
+{
+    private const CHARSET = '0123456789abcdfghijklmnpqrsvwxyz';
+
+    /**
+     * Nix-compatible hash compression.
+     */
+    public function compressHash(string $hash, int $size): string
+    {
+        return implode(
+            '',
+            array_map(
```

Comment:
I don't think doing this without intermediate variables improves readability. Ditto for the other functions.